src/cluster.c

```diff
@@ -1471,12 +1471,24 @@ void askingCommand(client *c) {
  * In this mode replica will not redirect clients as long as clients access
  * with read-only commands to keys that are served by the replica's primary. */
 void readonlyCommand(client *c) {
+    if (server.cluster_enabled == 0 && !(c->capa & CLIENT_CAPA_REDIRECT)) {
+        addReplyError(c, "This instance has cluster support disabled,"
+                         " you need to execute the CLIENT CAPA REDIRECT command,"
+                         " before you can use the READONLY command in standalone mode.");
```

Comment:
These very long error messages that include the full documentation are not my favourite style.

If you (a user) get an error, you would normally go to the documentation for the READONLY command and find out what it means.

A more concise way is to say that READONLY is allowed only if redirects are enabled. This includes cluster (always) and standalone (with client capa redirect).

```suggestion
        addReplyError(c, "Redirects are not enabled");
```